src/redis.c

```diff
@@ -140,11 +143,14 @@ static redis_query_t *redis_config_query(oconfig_item_t *ci) /* {{{ */
   for (int i = 0; i < ci->children_num; i++) {
     oconfig_item_t *option = ci->children + i;
 
+    rq->db = REDIS_DEF_DB;
     if (strcasecmp("Type", option->key) == 0) {
       status = cf_util_get_string_buffer(option, rq->type, sizeof(rq->type));
     } else if (strcasecmp("Instance", option->key) == 0) {
       status =
           cf_util_get_string_buffer(option, rq->instance, sizeof(rq->instance));
+    } else if (strcasecmp("Db", option->key) == 0) {
+      status = cf_util_get_int(option, &rq->db);
```

Comment:
Sorry, but I think we need to check option value to be positive here (so intermediate value will be needed).
What do you think?